text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,431 @@
+
+There is a potential to use the same metric names to describe SDK
+producers/consumers and Collector producers/consumers.  However, we
+find two reasons this unification step should be avoided.
+
+First, we seek to avoid aggregations combining first-class SDKs
+producers, SDK consumer components (i.e., bridges) and Collector
+producers, which do not have corresponding consumer metrics, and
+Collector producers.
+
+Second (really a specific case of the first), we seek to avoid
+aggregations that combine OpenTelemetry Collector pipeline metrics
+with SDK pipeline metrics in the same process, when the OpenTelemetry
+SDK instruments the OpenTelemetry Collector.
+
+### Collector outcomes are asymmetric
+
+There are a several of reasons why the producer and consumer outcomes
+counted by Collector pipeline monitoring will reflect contradictory
+information.
+
+For example, when timeouts are configured independently, as for
+example when a preceding segment's timeout is smaller than a following
+stage's timeout.  The preceding exporter's timeout, if less than the
+following exporter's timeout may cause consumer `timeout` outcomes
+without corresponding producer `timeout` outcomes.
+
+### Deferred outcomes
+
+In some configurations, Collector pipeline segments have asynchronous
+elements, in an arrangement where the `Consume()` operation called by
+the preceder on the follower returns success, and responsibility for
+delivery transfers to the follower.  When deferred outcomes are in
+use, consumer metrics will generally indicate 100% `accepted`
+outcomes.
+
+For these cases, exporters are expected to use the `deferred:` outcome
+categories to signal to monitoring systems especially the failure 
+outcomes that were not seen by producers.
+
+### Resource-exhausted case
+
+The `exhausted` outcome is meant to map to the gRPC RESOURCE_EXHAUSTED
+status code and the HTTP 429 status code.  These codes are special and
+distinct from the general-purpose `retryable` outcome because:
+
+- The failures they represent should generally be excluded in
+  calculating service-level objectives.  When these error codes are
+  returned, it is the caller's responsibility to remediate.
+- This outcome may lead to specifically different load-balancer
+  behavior, so it is an important signal for monitoring pipeline
+  health.
+
+## Pipeline equations
+
+The behavior of a pipeline section consisting of one or more elements
+can be reduced to distinct operation categories.
+
+The consumer categories, leading to the first pipeline segment equation:
+
+- **Received**: An item of telemetry was exported from a preceding pipeline segment
+- **Inserted**: An item of telemetry was inserted by this pipeline segment
+
+The first equation:
+
+```
+Consumed(Segment) == Recieved(Segment) + Inserted(Segment)
+```
+
+The producer categories, leading to the second pipeline segment equation:
+
+- **Exported**: An attempt was made to export the telemetry to a following pipeline segment
```

Comment:
is this an attempt or rather the data was successfully exported to the follower?

Reply:
An attempt. When the attempt is made, there is at least some expectation that the next pipeline segment has seen the data. Exported includes success and failed cases, and I'm not sure how I can change the words to improve this understanding. I mean to count cases where an RPC was made, essentially, whether it fails or not, because it sets up our expectation for the next segment.

Reply:
So, just to be clear, which metric do we use for an exporter that failed to even establish a connection to a downstream receiver?
For example, if I configure the collector with an OTLP exporter with a bad endpoint, and the HTTP/GRPC connection cannot be made, the export will "fail" but there is no expectation that any following receiver will ever see the data (so won't count it).
It seems `Exported` doesn't fit here by your definition.  Would it be `Dropped`?

Reply:
>It seems Exported doesn't fit here by your definition. Would it be Dropped?

Yes